lib/middleware/blp-session.ts

```diff
@@ -57,3 +54,48 @@ export function getSession(req: interfaces.IOurRequest,
             return next(new restify.InternalError(err.message));
         });
 }
+
+export function getSocketSession(socket: interfaces.ISocket) : Promise<interfaces.ISocket>
+{
+    return createSession(DEFAULT_ID)
+        .then((session: blpapi.Session): Promise<interfaces.ISocket> => {
+            socket.log.debug('blpSession retrieved.');
+            socket.blpSession = session;
+            return Promise.resolve(socket);
+        }).catch((err: Error): any => {
+            socket.log.error(err, 'Error getting blpSession.');
+            throw err;
+        });
+}
+
+/*
+export function getSession(req: interfaces.ISocket) : Promise<interfaces.ISocket>;
```

Comment:
Not sure which approach is better so I leave the function overloading implementation below. I personally think it is ugly and prefer the separate function way.